### ISSUE TITLE
[Problem] Arch - Crash seg fault when adding objects to Arch Project or Arch Site

### ISSUE THREAD

**Issue by marcuspollio** [FreeCAD/FreeCAD#12231]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Hi !

When using the latest "weekly", I noticed a segmentation fault leading to a crash.
It does not occur when using the latest stable release.

Please, try the following steps to reproduce :

1) Launch FreeCAD. In the Start WB Documents "New file" dialog pick "BIM/Architecture".
2) Model some basic geometric object, say a Wall.
3) With the object still selected (the Wall e.g.), add it to a Level (Arch_Floor) or Building container. No crash yet.
4) Select the Level or Building and add it to a Site or Project container. Crash !

Note there are many [recent opened issue](https://github.com/FreeCAD/FreeCAD/issues?q=is%3Aissue+is%3Aopen+crash) for similar seg fault crashes but none yet for Arch. Don't know how related to the others this issue is. Sorry if it's a duplicate. =D

Thanks for having a look !

### Full version info

Crash with:
```
OS: Ubuntu 22.04 LTS (GNOME)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.35759 (Git) AppImage
Build type: Release
Branch: main
Hash: 90d36f60c30cb6549a7987fcd9d4769eab1c160c
Python 3.10.13, Qt 5.15.8, Coin 4.0.2, Vtk 9.2.6, OCC 7.6.3
```
No crash with:
```
OS: Ubuntu 22.04 LTS (GNOME)
Word size of FreeCAD: 64-bit
Version: 0.21.2.33771 (Git) AppImage
Build type: Release
Branch: (HEAD detached at 0.21.2)
Hash: b9bfa5c5507506e4515816414cd27f4851d00489
Python 3.10.13, Qt 5.15.8, Coin 4.0.0, Vtk 9.2.6, OCC 7.6.3
```

### Subproject(s) affected?

Arch

### Anything else?

<details>

  <summary>Terminal log</summary>

```
Program received signal SIGSEGV, Segmentation fault.
#0  /lib/x86_64-linux-gnu/libc.so.6(+0x42520) [0x7f87c4e42520]
#1  /tmp/.mount_FreeCAZmyNdj/usr/lib/Part.so(+0x463af5) [0x7f86b2463af5]
#2  0x7f86b2473be2 in Part::TopoShape::makeCompound(std::vector<Part::TopoShape, std::allocator<Part::TopoShape> > const&, char const*, bool) from /tmp/.mount_FreeCAZmyNdj/usr/lib/Part.so+0x42
#3  /tmp/.mount_FreeCAZmyNdj/usr/lib/Part.so(+0x27a954) [0x7f86b227a954]
#4  0x7f86b227b822 in Part::Feature::getTopoShape(App::DocumentObject const*, char const*, bool, Base::Matrix4D*, App::DocumentObject**, bool, bool, bool) from /tmp/.mount_FreeCAZmyNdj/usr/lib/Part.so+0x1a2
#5  0x7f86b227bd08 in Part::Feature::getShape(App::DocumentObject const*, char const*, bool, Base::Matrix4D*, App::DocumentObject**, bool, bool) from /tmp/.mount_FreeCAZmyNdj/usr/lib/Part.so+0x48
#6  0x7f86ad3fb63f in PartGui::ViewProviderPartExt::updateVisual() from /tmp/.mount_FreeCAZmyNdj/usr/lib/./PartGui.so+0x10f
#7  0x7f86ad3feb8f in PartGui::ViewProviderPartExt::updateData(App::Property const*) from /tmp/.mount_FreeCAZmyNdj/usr/lib/./PartGui.so+0x7f
#8  0x7f87c8129430 in Gui::ViewProvider::update(App::Property const*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADGui.so+0x60
#9  0x7f87c813c847 in Gui::ViewProviderDocumentObject::update(App::Property const*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADGui.so+0x57
#10  0x7f87c7cec5ce in Gui::Document::slotChangedObject(App::DocumentObject const&, App::Property const&) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADGui.so+0x4e
#11  /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADApp.so(+0x218fb4) [0x7f87c7018fb4]
#12  0x7f87c7054012 in App::DocumentObject::onChanged(App::Property const*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADApp.so+0xd2
#13  0x7f87c7178905 in App::Property::hasSetValue() from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADApp.so+0x135
#14  0x7f87c71875a8 in App::PropertyContainerPy::setCustomAttributes(char const*, _object*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADApp.so+0x298
#15  0x7f87c7187c93 in App::PropertyContainerPy::_setattr(char const*, _object*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADApp.so+0x13
#16  0x7f86b227c572 in App::FeaturePythonPyT<Part::PartFeaturePy>::_setattr(char const*, _object*) from /tmp/.mount_FreeCAZmyNdj/usr/lib/Part.so+0x42
#17  0x7f86b226c596 in App::FeaturePythonPyT<Part::PartFeaturePy>::__setattro(_object*, _object*, _object*) from /tmp/.mount_FreeCAZmyNdj/usr/lib/Part.so+0x36
#18  /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libpython3.10.so.1.0(PyObject_SetAttr+0xfb) [0x7f87c752f1cb]
#19  /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libpython3.10.so.1.0(_PyEval_EvalFrameDefault+0xdc3) [0x7f87c753a703]
#20  /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libpython3.10.so.1.0(+0x155f91) [0x7f87c7555f91]
#21  0x7f87c70fe323 in App::FeaturePythonImp::execute() from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADApp.so+0x193
#22  0x7f86b2277a76 in App::FeaturePythonT<Part::Feature>::execute() from /tmp/.mount_FreeCAZmyNdj/usr/lib/Part.so+0x16
#23  0x7f87c7051f62 in App::DocumentObject::recompute() from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADApp.so+0xc2
#24  0x7f86b2277a1c in Part::Feature::recompute() from /tmp/.mount_FreeCAZmyNdj/usr/lib/Part.so+0xc
#25  0x7f87c70298fc in App::Document::_recomputeFeature(App::DocumentObject*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADApp.so+0x2fc
#26  0x7f87c7038dcd in App::Document::recompute(std::vector<App::DocumentObject*, std::allocator<App::DocumentObject*> > const&, bool, bool*, int) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADApp.so+0x5fd
#27  0x7f87c70b623e in App::DocumentPy::recompute(_object*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADApp.so+0x2ee
#28  0x7f87c70b6354 in App::DocumentPy::staticCallback_recompute(_object*, _object*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADApp.so+0x24
#29  /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libpython3.10.so.1.0(+0x149ba8) [0x7f87c7549ba8]
#30  /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libpython3.10.so.1.0(_PyObject_MakeTpCall+0x27b) [0x7f87c7542edb]
#31  /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libpython3.10.so.1.0(_PyEval_EvalFrameDefault+0x541e) [0x7f87c753ed5e]
#32  /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libpython3.10.so.1.0(+0x1560a2) [0x7f87c75560a2]
#33  /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADBase.so(PP_Run_Method+0xe3) [0x7f87c6bdc7a3]
#34  0x7f87c6b28f3b in Base::InterpreterSingleton::runMethodVoid(_object*, char const*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADBase.so+0x2b
#35  0x7f87c7d75dd3 in Gui::PythonCommand::activated(int) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADGui.so+0x53
#36  0x7f87c7d76468 in Gui::Command::_invoke(int, bool) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADGui.so+0x2a8
#37  0x7f87c7d76864 in Gui::Command::invoke(int, Gui::Command::TriggerSource) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADGui.so+0x134
#38  /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Core.so.5(+0x2e9214) [0x7f87c54e9214]
#39  0x7f87c636a422 in QAction::triggered(bool) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Widgets.so.5+0x32
#40  0x7f87c636cdeb in QAction::activate(QAction::ActionEvent) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Widgets.so.5+0xab
#41  /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Widgets.so.5(+0x25cb9a) [0x7f87c645cb9a]
#42  0x7f87c645ccd7 in QAbstractButton::mouseReleaseEvent(QMouseEvent*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Widgets.so.5+0xe7
#43  0x7f87c6545a4a in QToolButton::mouseReleaseEvent(QMouseEvent*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Widgets.so.5+0xa
#44  0x7f87c63aebd8 in QWidget::event(QEvent*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Widgets.so.5+0x178
#45  0x7f87c637035e in QApplicationPrivate::notify_helper(QObject*, QEvent*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Widgets.so.5+0x7e
#46  0x7f87c6378222 in QApplication::notify(QObject*, QEvent*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Widgets.so.5+0x13a2
#47  0x7f87c7d283f9 in Gui::GUIApplication::notify(QObject*, QEvent*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADGui.so+0x99
#48  0x7f87c54b3fc8 in QCoreApplication::notifyInternal2(QObject*, QEvent*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Core.so.5+0x108
#49  0x7f87c63763ab in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool, bool) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Widgets.so.5+0x1ab
#50  /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Widgets.so.5(+0x1c6e39) [0x7f87c63c6e39]
#51  /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Widgets.so.5(+0x1c9f99) [0x7f87c63c9f99]
#52  0x7f87c637035e in QApplicationPrivate::notify_helper(QObject*, QEvent*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Widgets.so.5+0x7e
#53  0x7f87c7d283f9 in Gui::GUIApplication::notify(QObject*, QEvent*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADGui.so+0x99
#54  0x7f87c54b3fc8 in QCoreApplication::notifyInternal2(QObject*, QEvent*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Core.so.5+0x108
#55  0x7f87c5b34bec in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Gui.so.5+0x84c
#56  0x7f87c5b0ad7c in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Gui.so.5+0xac
#57  /tmp/.mount_FreeCAZmyNdj/usr/plugins/platforms/../../lib/libQt5XcbQpa.so.5(+0x6454a) [0x7f87bf80c54a]
#58  /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/./libglib-2.0.so.0(+0x5a417) [0x7f87c3b11417]
#59  /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/./libglib-2.0.so.0(+0x5d3c7) [0x7f87c3b143c7]
#60  /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/./libglib-2.0.so.0(g_main_context_iteration+0x2e) [0x7f87c3b149be]
#61  0x7f87c55083d6 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Core.so.5+0x66
#62  0x7f87c54b2bbb in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Core.so.5+0x11b
#63  0x7f87c54ba6a2 in QCoreApplication::exec() from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libQt5Core.so.5+0x82
#64  0x7f87c7c8c25d in Gui::Application::runApplication() from /tmp/.mount_FreeCAZmyNdj/usr/bin/../lib/libFreeCADGui.so+0x20ad
#65  /tmp/.mount_FreeCAZmyNdj/usr/bin/freecad(+0x7620) [0x653b4a5ba620]
#66  /lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x7f87c4e29d90]
#67  /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80) [0x7f87c4e29e40]
#68  /tmp/.mount_FreeCAZmyNdj/usr/bin/freecad(+0x7981) [0x653b4a5ba981]
```

</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Syres916** [comment]
No crash or error using:

```
OS: Linux Mint 20.3 (X-Cinnamon/cinnamon)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.35806 (Git)
Build type: Release
Branch: main
Hash: 1bb5350c5102362889be02c14304eaefb5bee6c0
Python 3.8.10, Qt 5.12.8, Coin 4.0.0, Vtk 7.1.1, OCC 7.6.3
Locale: English/United Kingdom (en_GB)
Installed mods: 
  * freecad.gears 1.2.0
```

**Comment by marcuspollio** [comment]
Hi @Syres916 , thanks for having a look !

I just re-tested with the latest "weekly" (35803) AppImage and now the situation is even worse: it crashes as soon as I try to add an Arch object (Arch_wall, Arch_structure, etc.) =P

<details><summary>New FC infos</summary>

```
OS: Ubuntu 22.04 LTS (GNOME)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.35803 (Git) AppImage
Build type: Release
Branch: main
Hash: 90c90f705bec20faf4b8bb419574134c33c7d4b5
Python 3.10.13, Qt 5.15.8, Coin 4.0.2, Vtk 9.2.6, OCC 7.6.3
```
</details>

<details><summary>New log</summary>

```
Program received signal SIGSEGV, Segmentation fault.
#0  /lib/x86_64-linux-gnu/libc.so.6(+0x42520) [0x7b3f57c42520]
#1  0x7b3f5406727f in QQmlData::isSignalConnected(QAbstractDeclarativeData*, QObject const*, int) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/././libQt5Qml.so.5+0x1f
#2  /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Core.so.5(+0x2e8d67) [0x7b3f582e8d67]
#3  0x7b3f592b4ef9 in QWidgetLineControl::processKeyEvent(QKeyEvent*) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Widgets.so.5+0x119
#4  0x7b3f592a8faf in QLineEdit::keyPressEvent(QKeyEvent*) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Widgets.so.5+0x1f
#5  0x7b3f591af9ca in QWidget::event(QEvent*) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Widgets.so.5+0xf6a
#6  0x7b3f5917035e in QApplicationPrivate::notify_helper(QObject*, QEvent*) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Widgets.so.5+0x7e
#7  0x7b3f59177b9f in QApplication::notify(QObject*, QEvent*) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Widgets.so.5+0xd1f
#8  0x7b3f5ab27099 in Gui::GUIApplication::notify(QObject*, QEvent*) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libFreeCADGui.so+0x99
#9  0x7b3f582b3fc8 in QCoreApplication::notifyInternal2(QObject*, QEvent*) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Core.so.5+0x108
#10  /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Widgets.so.5(+0x1c9fe8) [0x7b3f591c9fe8]
#11  0x7b3f5917035e in QApplicationPrivate::notify_helper(QObject*, QEvent*) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Widgets.so.5+0x7e
#12  0x7b3f5ab27099 in Gui::GUIApplication::notify(QObject*, QEvent*) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libFreeCADGui.so+0x99
#13  0x7b3f582b3fc8 in QCoreApplication::notifyInternal2(QObject*, QEvent*) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Core.so.5+0x108
#14  0x7b3f58930b2b in QGuiApplicationPrivate::processKeyEvent(QWindowSystemInterfacePrivate::KeyEvent*) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Gui.so.5+0xdb
#15  /tmp/.mount_FreeCA3nFYqT/usr/plugins/platforminputcontexts/libibusplatforminputcontextplugin.so(+0xbebb) [0x7b3f51ad9ebb]
#16  /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Core.so.5(+0x2e9214) [0x7b3f582e9214]
#17  0x7b3f52490baf in QDBusPendingCallWatcher::finished(QDBusPendingCallWatcher*) from /tmp/.mount_FreeCA3nFYqT/usr/plugins/platforms/../../lib/libQt5DBus.so.5+0x2f
#18  0x7b3f582de190 in QObject::event(QEvent*) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Core.so.5+0x260
#19  0x7b3f5917035e in QApplicationPrivate::notify_helper(QObject*, QEvent*) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Widgets.so.5+0x7e
#20  0x7b3f5ab27099 in Gui::GUIApplication::notify(QObject*, QEvent*) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libFreeCADGui.so+0x99
#21  0x7b3f582b3fc8 in QCoreApplication::notifyInternal2(QObject*, QEvent*) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Core.so.5+0x108
#22  0x7b3f582b6c91 in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Core.so.5+0x181
#23  /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Core.so.5(+0x308cb3) [0x7b3f58308cb3]
#24  /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/./libglib-2.0.so.0(+0x5a417) [0x7b3f56bec417]
#25  /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/./libglib-2.0.so.0(+0x5d3c7) [0x7b3f56bef3c7]
#26  /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/./libglib-2.0.so.0(g_main_context_iteration+0x2e) [0x7b3f56bef9be]
#27  0x7b3f583083d6 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Core.so.5+0x66
#28  0x7b3f582b2bbb in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Core.so.5+0x11b
#29  0x7b3f582ba6a2 in QCoreApplication::exec() from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libQt5Core.so.5+0x82
#30  0x7b3f5aa8aefd in Gui::Application::runApplication() from /tmp/.mount_FreeCA3nFYqT/usr/bin/../lib/libFreeCADGui.so+0x20ad
#31  /tmp/.mount_FreeCA3nFYqT/usr/bin/freecad(+0x7620) [0x584b3cedd620]
#32  /lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x7b3f57c29d90]
#33  /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80) [0x7b3f57c29e40]
#34  /tmp/.mount_FreeCA3nFYqT/usr/bin/freecad(+0x7981) [0x584b3cedd981]
```
</details>

What can I do to help ? Thanks

**Comment by Roy-043** [comment]
I have created an issue for a similar problem (#12060). But have later closed it.

Please retest with "Allow aborting recomputation" checked.
And after that test again with that option unchecked.

**Comment by marcuspollio** [comment]
Hi @Roy-043 , thank you for having a look !

Ok, so I tested several times with this "Allow aborting recomputation" (Preferences>General>Document>General) unchecked and checked.

1) By default, when installing a fresh new instance or new AppImage, it is unchecked. FreeCAD crashes like reported in the OP (add an Arch object to Arch_Project).
2) With "Allow aborting recomputation", FreeCAD crashes as soon as a new Arch object is placed (3D view or Task panel), like mentioned in my 2nd message above.
3) When un-checking "Allow aborting recomputation", the situation is again like my 1) point.

I tested removing the .cache, .config and .local/share folders but it does not change anything.

BUT : I tested another workflow that does not lead to a crash.
 
1. Launch FreeCAD. In the Start WB Documents "New file" dialog pick "BIM/Architecture".
2. With an **empty** document, create an Arch_Project, deselect, create an Arch_Site, deselect, create an Arch_Building, deselect, create an Arch_Floor.
3. Arrange you hierarchy in the Model Tree panel (Project>Site>Building>Level).
4. Model some basic geometric object.
5. Move it to an Arch_Floor or Arch_Building via the Model Tree panel.

Before, I used the Toolbars or shortcuts leading to crashes.

Anything else I can test or help narrowing down the issue ? Thanks

**Comment by Roy-043** [comment]
#12269 may be related.

I have to test again. I see now that I did not follow your original scenario correctly.

**Comment by marcuspollio** [comment]
Hi @Roy-043 and @Syres916 ,

Good news : with commit [5d729db](https://github.com/FreeCAD/FreeCAD/commit/5d729db905aa9d74fc530be814a527dcd069fabf) applied, I can not make it crash anymore (with "Allow aborting recomputation" unchecked as per default).

Well done ! Thank you both for your support !

Closing for now.

<details><summary>Tested with latest AppImage</summary>
```
OS: Ubuntu 22.04 LTS (GNOME)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.35822 (Git) AppImage
Build type: Release
Branch: main
Hash: c57a83cb4f7f8bc8840ebbfc9aba97c15a502bd9
Python 3.10.13, Qt 5.15.8, Coin 4.0.2, Vtk 9.2.6, OCC 7.6.3
```
</details>

**Comment by marcuspollio** [comment]
Fixed by [5d729db](https://github.com/FreeCAD/FreeCAD/commit/5d729db905aa9d74fc530be814a527dcd069fabf)
